### PR TITLE
Add 'state' to MessagingExtensionAction

### PR DESCRIPTION
**Pull request by EricDahlvang**
Fixes https://github.com/microsoft/botbuilder-dotnet/issues/4709

### CODE REVIEW

**Comment by EricDahlvang** [comment]
This breaks our strict binary compat rules, but only due to a parameter addition.  Are we still enforcing that level?

File: libraries/Microsoft.Bot.Schema/Teams/Generated/MessagingExtensionAction.cs [hunk 1]
```diff
@@ -44,14 +44,16 @@ public MessagingExtensionAction()
         /// <param name="botActivityPreview">A collection of bot activities.</param>
         /// <param name="messagePayload">Message content sent as part of the
         /// command request.</param>
-        public MessagingExtensionAction(object data = default(object), TaskModuleRequestContext context = default(TaskModuleRequestContext), string commandId = default(string), string commandContext = default(string), string botMessagePreviewAction = default(string), IList<Activity> botActivityPreview = default(IList<Activity>), MessageActionsPayload messagePayload = default(MessageActionsPayload))
+        /// <param name="state">State parameter passed back to the bot after authentication flow.</param>
+        public MessagingExtensionAction(object data = default(object), TaskModuleRequestContext context = default(TaskModuleRequestContext), string commandId = default(string), string commandContext = default(string), string botMessagePreviewAction = default(string), IList<Activity> botActivityPreview = default(IList<Activity>), MessageActionsPayload messagePayload = default(MessageActionsPayload), string state = default(string))
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Schema/Teams/Generated/MessagingExtensionAction.cs:48]: This number of parameters is not recommended, and as we need more properties, the list will grow. We should introduce a method that receives an object containing all the data needed. If backward compat is an issue, maybe an overload?

Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Schema/Teams/Generated/MessagingExtensionAction.cs:48]: Agreed!  I think the plan is to create factory methods for this, and other Teams specific invoke responses.

That said, we do have a bunch of ridiculous and unruly constructors throughout the codebase: https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Schema/Activity.cs#L125

**Comment by mrivera-ms** [comment]
Need to turn off binary compat before we merge.

**Comment by BruceHaley** [comment]
Compatibility verifications can now be turned off by editing in ADO, BotBuilder-DotNet-CI-PR-yaml, setting the variable DisableApiCompatibityValidation to true. See PR #4761 for detail.

**Review by tomlm — APPROVED** [review]
:shipit:

File: libraries/Microsoft.Bot.Schema/Teams/Generated/MessagingExtensionAction.cs (hunk 1 above)
Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Schema/Teams/Generated/MessagingExtensionAction.cs]: Removed the new constructor parameter.

It is a shame we even have these constructors.

**Comment by stevengum** [comment]
Failing build has been requeued

**Comment by EricDahlvang** [comment]
I removed the new constructor parameter. so the build is passing normally.  


It would be great if we could change our binary compat library to do a compile compat check instead.  and/or even, remove all of these useless constructors someday.

**Review by mrivera-ms — APPROVED** [review]

**Pull request merged**